net/acme/files/run.sh

```diff
@@ -268,6 +270,16 @@ issue_cert()
     if [ -n "$dns" ]; then
         log "Using dns mode"
         acme_args="$acme_args --dns $dns"
+        if [ -n "$dalias" ]; then
+            log "Using domain alias for dns mode"
+            acme_args="$acme_args --domain-alias $dalias"
```

Comment:
Is this going to work when there's multiple domains? The examples on the wiki seem to indicate that the alias args need to go after the first domain definition - not sure if this is strictly a requirement, but if it is, your patch will end up with it in the wrong place...

Reply:
I read through the options parser and logic of the top-level acme.sh when changing this. Both the challenge alias and domain alias arguments go into a single ordered list (regardless of where they appear within the arguments overall).

I also tested this concept with multiple domains against both single and multiple alias options on my PC shell in various orders with no surprises. The only "odd" part is that the behavior for more domains than challenge/domain aliases currently does not use any alias. So if more than one alias is specified they must all be specified.

To avoid any issues I've simply limited the uci config to one of either challenge alias or domain alias so that only one element ends up in this list. That way acme.sh will apply it to all domains. If anything, the user can simply do one certificate per alias.

Let me know if you would still like it changed, but it would likely take some heavier refactoring.

Reply:
Nah, that's fine; just wanted to make sure you had checked :)